src/Microsoft.Data.Analysis/TextFieldParser.cs

```diff
@@ -0,0 +1,642 @@
+            {
+                return null;
+            }
+
+            ChangeBufferFunction BufferFunction = ReadToBuffer;
+            string line = ReadNextLine(ref _position, BufferFunction);
+            if (line == null)
+            {
+                FinishReading();
+                return null;
+            }
+
+            _lineNumber++;
+            return line.TrimEnd(newLineChars);
+        }
+
+        ///<summary>
+        /// Peek at <paramref name="numberOfChars"/> characters of the next data line without reading the line
+        ///</summary>
+        ///<param name="numberOfChars">The number of characters to look at in the next data line.</param>
+        ///<returns>A string consisting of the first <paramref name="numberOfChars"/> characters of the next line. >If numberOfChars is greater than the next line, only the next line is returned</returns>
+        public string PeekChars(int numberOfChars)
+        {
+            if (numberOfChars <= 0)
+            {
+                throw new ArgumentException($"{nameof(numberOfChars)} must be greater than 0");
+            }
+
+            if ((_reader == null) | (_buffer == null))
+            {
+                return null;
+            }
+
+            if (_endOfData)
+            {
+                return null;
+            }
+
+            string line = PeekNextDataLine();
+            if (line == null)
+            {
+                _endOfData = true;
+                return null;
+            }
+
+            line = line.TrimEnd(newLineChars);
+            if (line.Length < numberOfChars)
+            {
+                return line;
+            }
+
+            return line.Substring(0, numberOfChars);
+        }
+
+
+        public string ReadToEnd()
+        {
+            if ((_reader == null) | (_buffer == null))
+            {
+                return null;
+            }
+            StringBuilder builder = new StringBuilder(_buffer.Length);
+            builder.Append(_buffer, _position, _charsRead - _position);
+            builder.Append(_reader.ReadToEnd());
+            FinishReading();
+            return builder.ToString();
+        }
+
+        public void Close()
+        {
+            CloseReader();
+        }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                if (!_disposed)
+                {
+                    Close();
+                }
+                _disposed = true;
+            }
+        }
+
+        private void ValidateFieldTypeEnumValue(FieldType value, string paramName)
+        {
+            if (value < FieldType.Delimited || value > FieldType.FixedWidth)
+            {
+                throw new InvalidEnumArgumentException(paramName, (int)value, typeof(FieldType));
+            }
+        }
+
+        ~TextFieldParser()
```

Comment:
There are no unmanaged ressources, so the finalizer isn't needed.

Reply:
Ah, I thought I removed this. Artifact of using ILSpy to convert this code :). Fixed now